Tools/conversation_manager.py

```diff
@@ -18,29 +21,45 @@ def __new__(cls):
             cls._instance = super(ConversationManager, cls).__new__(cls)
         return cls._instance
 
+    @classmethod
+    def get_user_conversation(cls, telegram_user_id: int) -> Tuple['ConversationManager', str]:
+        conv_manager = cls()
+        user_id = UserCache().get_user_id(telegram_user_id)
+        return conv_manager, user_id
+
     def add_message(self, user_id: str, role: str, content: Union[str, Dict[str, Any], List[Dict[str, Any]]]) -> None:
+        if not user_id:
+            raise ValueError("User ID cannot be empty")
+
         if user_id not in self._conversations:
             self._conversations[user_id] = []
-        
+
         message = Message(
             role=role,
             content=content,
             timestamp=datetime.now()
         )
-        
+
         self._conversations[user_id].append(message)
-        
+
         if len(self._conversations[user_id]) > self._max_history:
             self._conversations[user_id] = self._conversations[user_id][-self._max_history:]
 
     def get_conversation_history(self, user_id: str) -> List[dict]:
-        messages = [
+        if not user_id:
+            raise ValueError("User ID cannot be empty")
+
+        if user_id not in self._conversations:
+            return []
+
+        return [
             {"role": msg.role, "content": msg.content}
             for msg in self._conversations[user_id]
         ]
-        
-        return messages
 
     def clear_history(self, user_id: str) -> None:
+        if not user_id:
```

Comment:
checking if user_id is null or empty is ok but it occurs 3 times. Error message can be done constants in static method:
```@staticmethod
def _validate_user_id(user_id: str) -> None:
    if not user_id:
        raise ValueError("User ID cannot be empty")```